### ISSUE TITLE
Ionic 4: expose access to native input element of ion-input

### ISSUE THREAD

**Issue by genadis** [ionic-team/ionic#15460]
# Feature Request

**Ionic Info**

```
Ionic:

   ionic (Ionic CLI)          : 4.1.1 (/usr/local/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.7
   @angular-devkit/core       : 0.7.3
   @angular-devkit/schematics : 0.7.3
   @angular/cli               : 6.1.3
   @ionic/ng-toolkit          : 1.0.6
   @ionic/schematics-angular  : 1.0.5

System:

   NodeJS : v8.9.3 (/usr/local/bin/node)
   npm    : 5.5.1
   OS     : macOS High Sierra
```

**Describe the Feature Request**
There are many use cases that cannot be all covered by ion-input so it makes sense to have access to the native input element to allow flexibility.

For example:
blur input, setSelectionRange and other standard apis, dealing with input of type "file" (accessing the files[] array), etc

At the moment The only solution I've found is 
```
@ViewChild('ionInputElRef', { read: ElementRef }) ionInputElRef: ElementRef;
```
And to access the native element:
```
const el = this.ionInputElRef.nativeElement.shadowRoot.querySelector('input') as HTMLInputElement;
```

**Describe Preferred Solution**
make nativeInput public property: https://github.com/ionic-team/ionic/blob/master/core/src/components/input/input.tsx#L17 

**Describe Alternatives**

**Related Code**

**Additional Context**

**Comment by njejina** [comment]
Please for Textarea too,,,

**Comment by calendee** [comment]
Maybe you're trying to access these events:

```
onInput={this.onInput}
onBlur={this.onBlur}
onFocus={this.onFocus}
onKeyDown={this.onKeydown}
```

https://github.com/ionic-team/ionic/blob/a9fd3399de84286adb44be0ca789e8ded67b9bc3/core/src/components/input/input.tsx#L366

If so, you can do so like this:

```
<ion-input onBlur={ this.onBlur } onInput={ this.onInput } name="username" type="text"></ion-input>
```

**Comment by calendee** [comment]
@njejina Textarea has the same : https://github.com/ionic-team/ionic/blob/a9fd3399de84286adb44be0ca789e8ded67b9bc3/core/src/components/textarea/textarea.tsx#L275

**Comment by calendee** [comment]
Here's a full sample:

```
import { Component, Prop, State } from '@stencil/core';

@Component({
  tag: 'app-profile',
  styleUrl: 'app-profile.css'
})
export class AppProfile {
  @State() state = false;
  @Prop() name: string;

  formattedName(): string {
    if (this.name) {
      return this.name.substr(0, 1).toUpperCase() + this.name.substr(1).toLowerCase();
    }
    return '';
  }

  onBlur(event): any {
    console.log('Blurred ', event.target.name);
  }

  onInput(event): any {
    console.log('Input ', event.target.name, event.target.value);
  }

  render() {

    return [
      <ion-header>
        <ion-toolbar color="primary">
          <ion-buttons slot="start">
            <ion-back-button defaultHref="/" />
          </ion-buttons>
          <ion-title>Profile: {this.name}</ion-title>
        </ion-toolbar>
      </ion-header>,

      <ion-content padding>
        <ion-item>
          <ion-label position="floating">Password</ion-label>
          <ion-input onBlur={ this.onBlur } onInput={ this.onInput } name="username" type="text"></ion-input>
        </ion-item>
      </ion-content>
    ];
  }
}
```

**Comment by RobotBroer** [comment]
I can't get this to work for me in Ionic 4.0.0-rc :
` 
if(this.input.nativeElement.shadowRoot === null) {
this.input.nativeElement.attachShadow({mode: 'open'})
const el = this.input.nativeElement.shadowRoot.querySelector("Input") as HTMLInputElement
console.log(el) }`
output of el is NULL 

I need access to the "HTMLInputElement since it is required to pass into Google Maps function: 
google.maps.places.Autocomplete(HTMLInputElement)

Now I am just stuck with the error, although function works as expected: 
> InvalidValueError: not an instance of HTMLInputElement

**Comment by EternalPhane** [comment]
This issue is already resolved by #17183.

**Comment by brandyscarney** [comment]
Thanks for the issue! Closing as fixed by https://github.com/ionic-team/ionic/pull/17183.
